Makefile

```diff
@@ -58,8 +58,15 @@ checksum:
 clean:
 	@rm -rf $(BINDIR) ./_dist
 
+GIT_BASEDIR    = $(shell git rev-parse --show-toplevel 2>/dev/null)
+ifneq ($(GIT_BASEDIR),)
+	LDFLAGS += -X github.com/Azure/acs-engine/pkg/test.JUnitOutDir=${GIT_BASEDIR}/test/junit
+endif
+
 test: test-style
-	go test -v $(GOFILES)
+	go get github.com/onsi/ginkgo/ginkgo
+	go get github.com/onsi/gomega
```

Comment:
the image used for distribution really shouldnt have our dev tools on it. Thats just added bloat and baggage we shouldnt have to worry about maintaining. The pattern most people follow is to have a development image that you can use for local and ci runs that contains all the necessary tooling while the image you build for distribution just has the binaries. 

Doing this has multiple benefits:

1 - it saves us from having to build an image during the CI run just to build binaries/run tests
2 - we reduce the image size we ship to our customers by not having unnecessary tooling installed
3 - when we need to update testing/development dependencies we dont have to cut an unnecessary release of a distribution image
4 - allows us to independently version the development image outside of the project so it can keep pace with tools as they come out (like updating ginkgo/gomega) but a project can stay pinned to a particular image version taht works for them until they are ready to upgrade.
5 - it means that for ANY given commit you will have the appropriate versioned development image that can build the corresponding binaries and run the tests. This allows someone who needs to build an older version of the project an easy way to do so without having to worry about installing old tools